Jobs.Fetcher.YouTube/Helpers/DbReader.cs

```diff
@@ -18,25 +18,53 @@ public static class DbReader {
         public static List<MetricDelta<Video, long>> CompareVideoLifetimeDailyTotal() {
             var now = DateTime.UtcNow;
             using (var analytics = new DataLakeYouTubeAnalyticsContext())
-                using (var db = new DataLakeYouTubeDataContext()) {
-                    var dayTotal =
-                        analytics.VideoDailyMetrics.Where(x => x.ValidityStart <= now && now < x.ValidityEnd)
-                            .GroupBy(x => x.VideoId, y => y, (k, v) => new { VideoId = k, Views = v.Sum(y => y.Views), StartDate = v.Min(y => y.Date), EndDate = v.Max(y => y.Date) }).ToList();
-                    var stats =
-                        from s in db.Statistics.Where(x => x.CaptureDate == now.Date && x.ValidityStart <= now && now < x.ValidityEnd)
-                        join dt in  dayTotal on s.VideoId equals dt.VideoId
-                        join v in db.Videos.Where(x => x.ValidityStart <= now && now < x.ValidityEnd) on s.VideoId equals v.VideoId
-                        select new MetricDelta<Video, long>(){
+            using (var db = new DataLakeYouTubeDataContext()) {
+
+                var dayTotal =
+                    analytics.VideoDailyMetrics
+                    .Where(x => x.ValidityStart <= now && now < x.ValidityEnd)
+                    .GroupBy(
+                        x => x.VideoId,
+                        y => y,
+                        (k, v) => new {
+                            VideoId = k, Views = v.Sum(y => y.Views),
+                            StartDate = v.Min(y => y.Date),
+                            EndDate = v.Max(y => y.Date) })
+                    .ToList();
+
+                var stats =
+                    from s in db.Statistics
+                    where
+                        s.CaptureDate == now.Date
+                        && now < s.ValidityEnd
+                    join dt in dayTotal
+                        on s.VideoId equals dt.VideoId
+                    join v in db.Videos
+                        .Where(x => x.ValidityEnd >= now)
+                        on s.VideoId equals v.VideoId
+                    select new MetricDelta<Video, long>(){
                         Id = v,
                         Lifetime = s.ViewCount,
                         Total = dt.Views,
                         DailyStart = dt.StartDate,
                         DailyEnd = dt.EndDate,
                         LifetimeDate = s.CaptureDate
-                    };
-                    return stats.ToList();
-                }
+                };
+                return stats.ToList();
```

Comment:
Sorry I had missed this message in the last review. I think the reason for this is that, if you don't, it returns an IQueriable, instead of a list. Might be what you want, in case you want lazy loading, then you can filter things before the request is actually sent to the DB. But in this case, I think it wouldn't matter

Maybe, we could remove `.ToList()` from `dayTotal`, that could make it just 1 trip to the DB instead of 2 (but I haven't really tested). But regarding the final result, the function consuming this would have to convert it to a list eventually. So, I think it is the same